### PR TITLE
Removed rpi specific config from the wpe layer

### PR DESCRIPTION
**Pull request by artur-gebicz-ml**
This is the first step in refactoring the meta-wpe layers.
Files and configuration that are RPI specific has been moved to the separate layer: `meta-metrological-raspberrypi.`
From this point onwards to build RaspberryPi image please use `meta-metrological-raspberrypi.`

### CODE REVIEW

File: conf/layer.conf [hunk 1]
```diff
@@ -32,23 +29,10 @@ LICENSE_PATH += "${LAYERDIR}/licenses"
 LICENSE_FLAGS_WHITELIST += "commercial_mpg123"
```
Comment (artur-gebicz-ml) [inline conf/layer.conf:29]: IMO: Those whitelists shouldn't be there too. We shouldn't whitelist the licences at all here

File: conf/layer.conf [hunk 2]
```diff
@@ -19,9 +19,6 @@ LAYERDEPENDS_wpe-layer = "core \
                          "
 
 BBFILES_DYNAMIC += " \
```
Comment (artur-gebicz-ml) [inline conf/layer.conf:21]: Not sure what to do with that. Let's keep that in mind for future

File: recipes-core/images/wpe-initramfs-image.bb [hunk 3]
```diff
@@ -5,5 +5,3 @@ DESCRIPTION = "WPE initramfs rootfs image"
 LICENSE = "MIT"
```
Comment (artur-gebicz-ml) [inline recipes-core/images/wpe-initramfs-image.bb:5]: This file and all images should be moved to the meta-metrological-rpi

File: recipes-graphics/westeros/westeros-sink.bbappend [hunk 4]
```diff
@@ -1,14 +1,9 @@
-SINK_SOC_PATH = "${@bb.utils.contains('MACHINE_FEATURES', 'vc4graphics', 'v4l2', 'rpi', d)}"
+SINK_SOC ?= ""
```
Comment (artur-gebicz-ml) [inline recipes-graphics/westeros/westeros-sink.bbappend:1]: I would use even weaker assignment( ??=). Let's keep that in mind when we will be adding new configuration options

File: recipes-graphics/westeros/westeros.bb [hunk 5]
```diff
@@ -2,10 +2,11 @@ require westeros.inc
 
 SUMMARY = "This receipe compiles the westeros compositor component"
 
-SRC_URI += "file://0001-Use-intptr_t-to-avoid-precision-errors-on-aarch64.patch \
+SRC_URI += " \
```
Comment (artur-gebicz-ml) [inline recipes-graphics/westeros/westeros.bb:5]: Adding file this way is convenient, but it causes warnings. Let's keep that in mind for later

File: recipes-wpe/wpebackend-rdk/wpebackend-rdk.inc [hunk 6]
```diff
@@ -25,15 +25,16 @@ def backendselector(d):
     compositor = bb.utils.contains('DISTRO_FEATURES', 'compositor', True, False, d)
 
     if compositor == False and wayland == False:
-        return 'rpi'
+        return 'platform'
```
Comment (artur-gebicz-ml) [inline recipes-wpe/wpebackend-rdk/wpebackend-rdk.inc:28]: Can we have this fix as a separate commit? Just not to mix fixes with refactoring

File: recipes-wpe/wpeframework/include/playerinfo.inc [hunk 7]
```diff
@@ -2,11 +2,11 @@
 WPE_PLAYERINFO_AUTOSTART ?= "true"
 WPE_PLAYERINFO_DOLBY_FLAGS = "${@bb.utils.contains('DISTRO_FEATURES', 'dolby', '-DDOLBY_SUPPORT=ON', '-DDOLBY_SUPPORT=OFF', d)}"
 WPE_PLAYERINFO_DOLBY_PLATFORM_IMPL ?= "None"
-DEPENDS_append_rpi = " gstreamer1.0"
+WPE_PLAYERINFO_DEP ?= ""
```
Comment (artur-gebicz-ml) [inline recipes-wpe/wpeframework/include/playerinfo.inc:5]: I would use even weaker assignment( ??=). Let's keep that in mind when we will be adding new configuration options

File: recipes-wpe/wpeframework/include/webkitbrowser.inc [hunk 8]
```diff
@@ -16,9 +16,7 @@ WEBKITBROWSER_WEBINSPECTOR ?= "0.0.0.0:9998"
 WEBKITBROWSER_USER ?= ""
 WEBKITBROWSER_GROUP ?= ""
 WEBKITBROWSER_RESOLUTION ?= "720p"
-
-# Raspberry PI 3 has 4 cores, set to 4 for RPIs
-WEBKITBROWSER_THREADEDPAINTING_rpi ?= "2"
+WEBKITBROWSER_THREADEDPAINTING ?= ""
```
Comment (artur-gebicz-ml) [inline recipes-wpe/wpeframework/include/webkitbrowser.inc:19]: I would use even weaker assignment( ??=). Let's keep that in mind when we will be adding new configuration options

File: recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb [hunk 9]
```diff
@@ -1,13 +1,14 @@
 SUMMARY = "WPEFramework rdkservices"
```
Comment (artur-gebicz-ml) [inline recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb:1]: Can we make a bump a separate commit?

**Review by artur-gebicz-ml — COMMENTED** [review]
Here are my remarks. For now let's leave _brcm to no break compatibility, but we need to think how to do that.
Also we need to be 100% that everything that was removed here will be added to the new layer

File: recipes-core/images/wpe-initramfs-image.bb (hunk 3 above)
Comment (HaseenaSainul) [reply recipes-core/images/wpe-initramfs-image.bb:5]: Do we need to move all image files ? then we have to create this image file for each platform right ?

File: recipes-graphics/westeros/westeros.bb (hunk 5 above)
Comment (HaseenaSainul) [reply recipes-graphics/westeros/westeros.bb:5]: I am not seeing any warning due to this. will revert, if I get it

File: recipes-wpe/wpebackend-rdk/wpebackend-rdk.inc (hunk 6 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpebackend-rdk/wpebackend-rdk.inc:28]: done

File: conf/layer.conf (hunk 1 above)
Comment (HaseenaSainul) [reply conf/layer.conf]: moved to platform layer

File: recipes-graphics/westeros/westeros-sink.bbappend (hunk 4 above)
Comment (HaseenaSainul) [reply recipes-graphics/westeros/westeros-sink.bbappend]: done

File: recipes-wpe/wpeframework/include/playerinfo.inc (hunk 7 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/include/playerinfo.inc]: done

File: recipes-wpe/wpeframework/include/webkitbrowser.inc (hunk 8 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/include/webkitbrowser.inc]: done

File: recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb (hunk 9 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/wpeframework-rdkservices_git.bb:1]: will do after do the settings/cleanup of thunder modules and plugins

File: recipes-core/images/wpe-initramfs-image.bb (hunk 3 above)
Comment (artur-gebicz-ml) [reply recipes-core/images/wpe-initramfs-image.bb:5]: Correct. No one exept rpi builds use this. meta-wpe should be platform independent, so please remove them here and add them to the meta-metrological-raspberrypi

Comment (HaseenaSainul) [reply recipes-core/images/wpe-initramfs-image.bb]: removed from this layer. it was already added to ml rpi layer

**Review by HaseenaSainul — APPROVED** [review]
